test_provider.py

```diff
@@ -0,0 +1,310 @@
+        print('observationId: ', json['observationId'])
+        atoms = JsonProvider.parse_atoms(json['sequence'], qa_states)
+
+        obs = Observation(json['observationId'],
+                          json['key'],
+                          int(name.split('-')[1]),
+                          json['title'],
+                          site,
+                          status,
+                          None,
+                          priority,
+                          SetupTimeType(json['setupTimeType']),
+                          timedelta(milliseconds=json['setupTime']),
+                          JsonProvider.obs_classes[json['obsClass']],
+                          None,
+                          None,
+                          None,
+                          targets,
+                          guide_stars,
+                          atoms,
+                          constraints,
+                          None
+                          )
+
+
+        # obs.active = ...
+
+        # obs.exec_time = ... # Total time sequence  + overhead
+        # obs.program_used = ...
+        # obs.partner_used = ...
+       
+        # obs.too_type = ... # TODO: at program level
+
+        return obs
+    
+    @staticmethod
+    def parse_time_allocation(json: dict) -> TimeAllocation:
+        #print(json['timeAccountAllocationCategories'])
+        ta = TimeAllocation(TimeAccountingCode(json['timeAccountAllocationCategories'][0]['category']),
+                            timedelta(milliseconds=json['awardedTime']),
+                            timedelta(milliseconds=0),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['programTime']),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['partnerTime']))
+        return ta
+
+    @staticmethod
+    def parse_or_group(json: dict) -> OrGroup:
+        
+        # Find nested OR groups/AND groups
+        # TODO: is this correct if there are not nested groups in OCS natively
+
+        observations = [JsonProvider.parse_observation(json[key], key) for key in json.keys() if key.startswith('OBSERVATION_BASIC')]
+
+        number_to_observe = len(observations)
+        delay_max, delay_min = 0, 0 # TODO: What are these?
+        or_group = OrGroup(json['key'], json['name'], number_to_observe, delay_min, delay_max, observations)
+        return or_group
+
+    @staticmethod
+    def parse_guide_star(json: dict) -> Resource:
+
+        # TODO: Maybe a GuideStart class should be in place to handle this better
+        res = Resource(json['name'], json['tag'])
+        return res
+    
+    @staticmethod
+    def parse_root_group(json: dict) -> OrGroup:
+        # Find nested OR groups/AND groups
+        groups = [JsonProvider.parse_and_group(json[key]) for key in json.keys() if key.startswith('GROUP_GROUP_SCHEDULING')]
```

Comment:
Just a note: we may have to handle `ORGANIZATIONAL_FOLDER` as well, which I think would be the same as a scheduling group, but I'll have to test.

Reply:
Here's the definitive reply from the Observing Tool: an organizational folder is just for human convenience and holds no relevance, so when you encounter an organizational folder, we could either just roll it up to the parent group and include the observations in it there as `AndGroup`s containing the `Observation`s, or create an `AndGroup` and create sub `AndGroup`s with the `Observation`s in them.
<img width="741" alt="Screen Shot 2021-12-19 at 8 51 25 PM" src="https://user-images.githubusercontent.com/8795653/146724334-66627901-7728-471b-9000-6e1f887e60f0.png">